components/invoice-upload.tsx

```diff
@@ -0,0 +1,189 @@
+        } else {
+          const errorText = await response.text();
+          setMessage(`File upload failed: ${errorText}`);
+          return;
+        }
+      }
+      setUploadedFiles(uploadedFilesDetails); // Replace previous files with new ones
+      setMessage(`Files successfully uploaded.`);
+      setFiles([]);
+      if (fileInputRef.current) fileInputRef.current.value = ''; // Reset file input
+    } catch (error) {
+      console.error('Error uploading file:', error);
+      setMessage('File upload failed due to a network or server error.');
+    }
+  };
+
+  const handleAccept = async () => {
+    const API_SERVER_URL = process.env.API_SERVER_URL;
+    const API_CLIENT_SECRET = process.env.API_CLIENT_SECRET;
+
+    try {
+      const response = await fetch(`${API_SERVER_URL}/api/invoice/accept`, {
+        method: 'POST',
+        headers: {
+          'Authorization': `Bearer ${API_CLIENT_SECRET}`
+        }
+      });
+
+      if (response.ok) {
+        setMessage('Invoice accepted successfully.');
+      } else {
+        const errorText = await response.text();
+        setMessage(`Accept action failed: ${errorText}`);
+      }
+    } catch (error) {
+      console.error('Error accepting invoice:', error);
+      setMessage('Accept action failed due to a network or server error.');
+    }
+  };
+
+  const handleReject = () => {
+    // Clear the state
+    setFiles([]);
+    setMessage('');
+    setUploadedFiles([]);
+    if (fileInputRef.current) fileInputRef.current.value = ''; // Reset file input
+  };
+
+  return (
+    <div className="flex flex-col items-center justify-start w-full h-screen bg-gray-100 px-4 py-8">
+      <div className="w-full max-w-7xl mx-auto bg-white rounded-lg shadow-lg p-8">
+        <h1 className="text-3xl font-bold mb-6 text-center">Invoice Upload</h1>
+        <form onSubmit={handleSubmit} className="space-y-4">
+          <div className="flex justify-center items-center bg-gray-50 p-4 rounded">
+            <input 
+              ref={fileInputRef}
+              type="file" 
+              multiple
+              onChange={handleFileChange} 
+              className="block w-full text-sm text-gray-500 file:mr-4 file:py-2 file:px-4 file:rounded file:border file:border-gray-300 file:text-sm file:font-semibold file:bg-blue-50 file:text-blue-700 hover:file:bg-blue-100"
+            />
+          </div>
+          <button 
+            type="submit" 
+            className="w-full bg-blue-600 hover:bg-blue-700 text-white py-2 px-4 rounded transition-colors duration-150"
+          >
+            Upload
+          </button>
+        </form>
+        {message && (
+          <div className="mt-4 text-center text-sm text-gray-600">
+            {message}
+          </div>
+        )}
+        {uploadedFiles.length > 0 && (
+          <div className="overflow-x-auto mt-6">
+            <table className="min-w-full text-sm text-left text-gray-500">
```

Comment:
We don't need multiple rows here since we have only one invoice. I think we can use the available freely to show the details related to the given invoice.

Reply:
done!